packages/react/src/primitives/TextField/TextField.tsx

```diff
@@ -59,25 +62,29 @@ export const TextField: React.FC<TextFieldProps> = ({
         labelHidden={labelHidden}
         descriptiveText={descriptiveText}
       />
-      <Input
-        aria-labelledby={fieldId}
-        defaultValue={defaultValue}
-        hasError={hasError}
-        id={fieldId}
-        isDisabled={isDisabled}
-        isReadOnly={isReadOnly}
-        isRequired={isRequired}
-        size={size}
-        type={type}
-        value={value}
-        onChange={onChange}
-        onCopy={onCopy}
-        onCut={onCut}
-        onInput={onInput}
-        onPaste={onPaste}
-        onSelect={onSelect}
-        {...rest}
-      />
+      <FieldGroup>
+        <Input
+          aria-labelledby={fieldId}
+          className={ComponentClassNames.FieldGroupField}
+          defaultValue={defaultValue}
+          hasError={hasError}
+          id={fieldId}
+          isDisabled={isDisabled}
+          isReadOnly={isReadOnly}
+          isRequired={isRequired}
+          onChange={onChange}
+          onCopy={onCopy}
+          onCut={onCut}
+          onInput={onInput}
+          onPaste={onPaste}
+          onSelect={onSelect}
+          size={size}
+          type={type}
+          value={value}
+          {...rest}
+        />
+        {inputEndComponent}
```

Comment:
For balance sake, should we also add `inputStartComponent`? Also, would this work if `inputEndComponent` is actually multiple React nodes?

Reply:
Yes, I think we should add inputStartComponent as well.

Do you have an example of where we would want more than one end component? I see that you can do something similar with [bootstrap inputgroups](https://getbootstrap.com/docs/4.0/components/input-group/#multiple-addons). Perhaps we could take a similar approach to support multiple nodes.

Reply:
Updated to add `inputStartComponent`, and moved logic into the `FieldGroup` component.